src/clarity/diagnostic.rs

```diff
@@ -71,55 +71,55 @@ impl Diagnostic {
         } else {
             output.push(format!("{}: {}", self.level, self.message,));
         }
-        if self.spans.len() > 0 {
-            let span = &self.spans[0];
+        output.append(&mut self.output_code(lines));
+        output
+    }
+
+    pub fn output_code(&self, lines: &Vec<String>) -> Vec<String> {
+        let mut output = Vec::new();
+        if self.spans.is_empty() {
+            return output;
+        }
+        let span = &self.spans[0];
+        let first_line = span.start_line.saturating_sub(1) as usize;
+        let last_line = span.end_line.saturating_sub(1) as usize;
+
+        output.push(lines[first_line].clone());
+        let mut pointer = format!(
+            "{: <1$}^",
+            "",
+            (span.start_column.saturating_sub(1)) as usize
+        );
+        if span.start_line == span.end_line {
+            pointer = format!(
+                "{}{:~<2$}",
+                pointer,
+                "",
+                (span.end_column - span.start_column) as usize
+            );
+        }
+        pointer = format!("{}", pointer);
+        output.push(pointer);
+
+        for span in self.spans.iter().skip(1) {
             let first_line = span.start_line.saturating_sub(1) as usize;
             let last_line = span.end_line.saturating_sub(1) as usize;
 
             output.push(lines[first_line].clone());
-            let mut pointer = format!(
-                "{: <1$}^",
-                "",
-                (span.start_column.saturating_sub(1)) as usize
-            );
+            let mut pointer = format!("{: <1$}^", "", (span.start_column - 1) as usize);
```

Comment:
What is the reason for this change from `saturating_sub(1)` to `- 1`. I think it was using the saturating sub so that it correctly handles the case where the start_column is 0.

Reply:
Addressed with [latest](https://github.com/hirosystems/clarity-repl/pull/159/commits/257b12af8673c56d8112232e2489b564beaf5755)